### ISSUE TITLE
Kafka output doesn't report failure when non-seed brokers are blocked by a firewall

### ISSUE THREAD

**Issue by faec** [elastic/beats#33206]
- Version: 8.1.3
- Operating System: Linux
- Steps to Reproduce:
  * Create a Kafka cluster with two nodes and create an unreplicated 1-partition topic. Check which Kafka node the topic is hosted on.
  * Configure the Filebeat Kafka output to write to the Kafka topic. Set its `hosts` key to contain only the node that is _not_ hosting the topic.
  * Configure a firewall to _silently_ drop connections between Filebeat and the Kafka topic node (i.e. to let attempted connections hang in `SYN_SENT` rather than responding with failure / `RST`)

This will (as expected) prevent ingestion into the topic, which will eventually stall Filebeat's pipeline entirely. However, Filebeat does not log an error the way it would with a typical failed connection or seed broker error, but continues to silently retry indefinitely.

**Comment by aveuiller** [comment]
Hello,

As part of this bug resolution, it would be great to add some metrics on the state of Filebeat.
Namely, metrics that I would love to see under the `/stats` path, would be:
- The last registry update timestamp
- The number of Kafka output brokers by connection state (_e.g._ pending, connected, in error).

Those metrics did help during the investigation of the issue, and it could be very handy to have them natively generated by Filebeat.

Sorry if that's not the right place to ask for this, feel free to move the query in another issue if that feels more relevant.

Thank you!

**Comment by aveuiller** [comment]
👍
